include/lang_utils/tuple.h

```diff
@@ -132,9 +143,10 @@ auto map_tuple_p(FUNC &&func, TUPLES&& ...tuples) {
 
 
 template <size_t LAST_INDEX, typename FUNC, typename ...TUPLES>
-class reduce_tuple_helper {
-    template <typename F, typename S, typename ...T>
-    friend auto reduce_tuple(F&&, S&&, T&&...);
+struct reduce_tuple_helper {
+    // clang doesn't like template friends.
+    // template <typename F, typename S, typename ...T>
+    // friend auto reduce_tuple(F&&, S&&, T&&...);
```

Comment:
I dislike that many recent projects use and target clang. GCC has nice features like more verbose (which sometimes means more useful) template error messages, and earlier support like some extensions (string literal operator templates, template syntax for lambdas, concepts).